packages/nimble-components/src/table/index.ts

```diff
@@ -300,6 +314,26 @@ export class Table<
         });
     }
 
+    public async setRecordHierarchyOptions(
+        rowHierarchyOptions: {
+            id: string,
+            options: TableRecordHierarchyOptions
+        }[]
+    ): Promise<void> {
+        await this.processPendingUpdates();
+
+        if (!this.isHierarchyEnabled()) {
+            // Do not allow hierarchy options to be configured when hierarchy is not
+            // configured on the table.
+            return;
```

Comment:
Perhaps my intention in the first bullet point you mentioned would have been better captured by saying _all options will be **discarded** if parentIdFieldName is not configured_. The goal was to keep a client from configuring hierarchy options if the table wasn't in hierarchy mode, but not to throw existing configuration out in case a binding did something odd like switch from a defined `parentIdFieldName` to `undefined` and back. That being said, if we think clients will run into binding-related timing issues then perhaps we should do what you're suggesting and hold on to their updated hierarchy options.

@rajsite, any opinion one way or another here?

Reply:
> That being said, if we think clients will run into binding-related timing issues then perhaps we should do what you're suggesting and hold on to their updated hierarchy options.

Yea the discussed goal was to minimize the coupling to binding order. Users do need to worry about it for idFieldName but they don't need to worry about it for parentIdFieldName. So we should save whatever they specify for hierarchy state regardless if hierarchy mode is on or off at the moment. It's state that applies to the records so it only should shift around if the records change.

Reply:
I've updated the implementation, and I've also removed the bullet point that says _all options will be ignored if parentIdFieldName is not configured on the table_ since that behavior is already described in the other bullet point @atmgrifter00  mentioned (_will not render delayed hierarchy state (loading or expandable) if the table's parentIdFieldName is not configured; however, the options will remain cached within the table if the parentIdFieldName becomes undefined, and that cached configuration will render in the table if the table's parentIdFieldName is changed back to a non-undefined value_).